config/confighttp/confighttp.go

```diff
@@ -61,7 +62,8 @@ type HTTPClientSettings struct {
 }
 
 // ToClient creates an HTTP client.
-func (hcs *HTTPClientSettings) ToClient(ext map[config.ComponentID]component.Extension) (*http.Client, error) {
+func (hcs *HTTPClientSettings) ToClient(
+	ext map[config.ComponentID]component.Extension, defaultUserAgent string) (*http.Client, error) {
```

Comment:
Changing the interface here doesn't feel right since the rest of the configuration happens by setting values on the `HTTPClientSettings` struct.  What about changing `addDefaultUserAgent()` to an exported method on that struct that can be invoked before `ToClient()`?

Reply:
Updated in the new commit